### PR TITLE
feat: Output fsLR meshes on subject surfaces

### LINKED ISSUE #447: Write out fsnative surface files registered to fsLR space
**Is your feature request related to a problem? Please describe.**

My understanding is that the resulting surface files will have the same vertices and mesh as the standard-space (dhcpAsym or fsLR) surface, but will retain the subject's fsnative geometry. This way users can visualize standard-space surface maps on the subject's unique topology (sulci, gyri, etc.), as is recommended in [Jeganathan et al. (preprint)](https://doi.org/10.1101/2024.07.09.602799).

**Describe the solution you'd like**

sMRIPrep would output pial, white matter, midthickness, inflated, and very-inflated surface files with `space-fsLR`.

**Describe alternatives you've considered**

Currently, XCP-D performs this step, but I'm pretty sure the nipreps team and DCAN lab have discussed it and wanted to move it into sMRIPrep/fMRIPrep/Nibabies.

**Additional context**

This is a duplicate of https://github.com/nipreps/nibabies/issues/367.

### PR DESCRIPTION
**Pull request by effigies**
Closes #447.

### CODE REVIEW

**Comment by effigies** [comment]
@tsalo Could I bother you for a review?

**Review by tsalo — APPROVED** [review]
This looks awesome!

The looping in `init_resample_surfaces_wf` goes a little over my head, but everything looks great.

One question- would it make sense to calculate the inflated and veryinflated surfaces in fsLR space, or would you prefer to leave that to the post-processing workflow?

**Comment by effigies** [comment]
> One question- would it make sense to calculate the inflated and veryinflated surfaces in fsLR space, or would you prefer to leave that to the post-processing workflow?

I don't have a strong preference not to do it, but I also don't want to hold up this feature on my understanding another. Feel free to open a follow-up issue.

**Pull request merged**
